### ISSUE TITLE
Support aria-labelledby in ComposeDialog in order to handle accessibility violation

### ISSUE THREAD

**Issue by ghost** [carbon-design-system/carbon#7824]
This is a follow-up for https://github.com/carbon-design-system/carbon/issues/6543.
The current fix introduced support for aria-label and some computation to re-use label from ModalHeader.

## Environment

MacOS, Chrome

> Assistive technology used to verify
IBM Equal Access Accessibility Checker

## Detailed description

> What version of the Carbon Design System are you using?

"carbon-components": "10.28.0",
"carbon-components-react": "7.27.0",

> What did you expect to happen?

One of possible solutions for "Interactive component does not have a programmatically associated name" is "Use the aria-labelledby attribute to reference existing visible text on the page that can clearly identify the purpose of the component"

## Steps to reproduce the issue

Steps are the same as for https://github.com/carbon-design-system/carbon/issues/6543
Setting aria-labelledby on ComposeDialog does not fix the violation.

**Comment by ghost** [comment]
I have PR created. I am sending it for the review in couple minutes.

**Comment by tw15egan** [comment]
@szymonbrandys do you have a link to a reproducible example that is throwing this a11y violation?

**Comment by tw15egan** [comment]
https://codesandbox.io/s/zealous-varahamihira-z6q37?file=/src/index.js
